string type
int32 num
---
string is_detect
